📦 NEW: Support For Multiple Cities in Pakistan

mian-muhammad opened the pull request:
Added support for almost any city of Pakistan. Instead of importing data from json files, scrapping it from urdupoint's website.

mian-muhammad commented:
Sure, will look into that

mian-muhammad commented:
@ahmadawais changes made. Scrapped data and updated the data directory.

mian-muhammad commented:
I am looking into Geo's website, there few things I want to say:
1. There is no consistency in h2's
2. They are only providing data for 23 cities

Where as on [Urdupoint](https://www.urdupoint.com/islam/pakistan-ramadan-calendar-sehar-aftar-timings.html) there data is quite consistent for scraping and they providing data for around 200 cities which includes KPK's cities. Also in KPK's few cities first Ramadan was on 24th April. This might be the reason few data files were of 24th insted of 25th April.

![image](https://user-images.githubusercontent.com/20898558/81729890-2be2fe80-94a6-11ea-87a0-471f124ed2f1.png)

mian-muhammad commented:
@ahmadawais also if you allow me I can post scraper for generating data but that is made on [urdupoint's website](https://www.urdupoint.com/islam/pakistan-ramadan-calendar-sehar-aftar-timings.html)

ahmadawais commented:
I don't want the GEO site. Your data is wrong. I don't care what source you use as far as the data is correct. I mentioned Geo as a reference. Please read all of my comment again. Add the script that scrapes and remove the data files. Also the scrapper should produce wrote data otherwise the roza number would be wrong.

mian-muhammad commented:
oh my bad, actually there 29 entities for Ramadan calendar on the source I was using, that's why scraper was creating 29 days instead of 30.

ahmadawais commented:
Then find a way to fix it, use another source, find an API if you can. But in the current form, the data that gets generated is incomplete and wrong. Which makes it difficult for me to accept this PR despite all the time and effort you put into this.

mian-muhammad commented:
I am looking into that @ahmadawais

ahmadawais commented on package.json:
```diff
@@ -33,10 +33,13 @@
+		"get-data": "node ./utils/generateData.js"
 	},
 	"dependencies": {
+		"axios": "^0.19.2",
```
This is a `devDependency`

ahmadawais commented on package.json:
```diff
@@ -33,10 +33,13 @@
 	"dependencies": {
+		"axios": "^0.19.2",
 		"chalk": "^4.0.0",
+		"cheerio": "^1.0.0-rc.3",
```
This is a `devDependency`

ahmadawais commented on readme.md:
```diff
@@ -11,7 +11,7 @@
 - 🌜 Get Sehar/Iftar times for your city
 - 🗓 Get Sehar/Iftar times for entire month of Ramadan
-- 🇵🇰 10+ major cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`
+- 🇵🇰 99 cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`
```
```suggestion
- 🇵🇰 99 cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`, and many more
```

mian-muhammad commented:
@ahmadawais 

- Data files removed

- Added scraper which scrapes data from [hmariweb](https://hamariweb.com/islam/pakistan_ramadan-timings1.aspx?Page=1), for 99 cities.

- Added an npm script `npm run get-data` which will generate data files for 99 cities replacing the old ones and will also update `cities.js` file.

- Made minor changes in `readme.md`

- Support for multi word cities

ahmadawais commented on index.js:
```diff
@@ -17,11 +17,13 @@ const notFound = `${sym.error} ${input}: not found.\n Consider contributing!`;
-	const city = input ? input.toLowerCase() : `lahore`;
+	const city = input ? input.toLowerCase().replace(/\s+/g, '-') : `lahore`;
+	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1).replace(/-/g, ' ')}`;
+	console.log(cityNiceName);
```
Don't log anything.

ahmadawais commented on utils/theEnd.js:
```diff
@@ -1,8 +1,8 @@
-module.exports = async ({city, noData}) => {
-	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1)}`;
+module.exports = async ({ city, noData }) => {
+	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1).replace(/-/g, ' ')}`;
```
This should have the `cityNiceName` otherwise where is it used?

ahmadawais commented:
@mian-muhammad the problem I mentioned is the same. You didn't address it. I'm currently checking the data for Lahore and it's wrong. The first ramadan is shown as 24th April and 30th of Ramadan is shown as 23rd May — this is all wrong. 

When I run `npx ramadan-cli` it shows me the wrong data of 12th May instead of 13th May that would be today because you are adding an extra roza in the start which is wrong.

The idea is great. I love the work you have done here. But I can't use it if the data produced by this work is wrong. Can you fix this issue for all cities and request a review once that's done?

mian-muhammad commented:
@ahmadawais sorry I was not aware of the behavior of toISOString, that's why dates were messed up. I have updated the code as you asked:

- Added `cheerios` and `axios` as `devDependencies`

- Removed additional log statements 

- Fixed the date issue

- Update `readme.md` file

ahmadawais commented:
@mian-muhammad I'm sorry I was not able to entertain your PR. Did you end up fixing the data?
